api/HttpClient.ts

```diff
@@ -146,14 +147,45 @@ export class HttpClient implements ifm.IHttpClient {
     }
 
     request(protocol: any, options: any, objs: any, onResult: (err: any, res: http.ClientResponse, contents: string) => void): void {
+        // Set up a callback to pass off 401s to an authentication handler that can deal with it
+        let self = this;
```

Comment:
you can avoid this if you use a fat-arrow function in the callback below:

    var callback = (err, res, contents) => {
      // replace self with this
    });`